core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -171,12 +188,13 @@ public static Socket connect(Properties props, String socketPathFormat) throws I
             + "connection Properties with value in form \"project:region:instance\"");
 
     // GAE Standard + GCF provide a connection path at "/cloudsql/<CONNECTION_NAME>"
-    if (forceUnixSocket || runningOnGaeStandard() || runningOnGoogleCloudFunctions()) {
+    String unixSocket = props.getProperty(UNIX_SOCKET_PROPERTY);
```

Comment:
It looks like all the callers to `migrateForceSocketVarToProperty(props)` immediately call this method `connect(props)`. So why not do the migration here, once? And then you don't even need to mutate the properties. You could just have a method like `getUnixSocket(props)` that returns the unix socket property, or the /cloudsql/instance string if the environment variable is set, and appends the suffix if needed.

```java
String unixSocket = getUnixSocket(props);
if (unixSocket != null) {
  logger.info(…);
  return UnixSocketChannel.open…
}
```

Reply:
So my goal was to try and encapsulate the Postges specific logic inside the Postgres module. It unfortunately has the suffix that has to be added:

https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/6579df05e078a0f76241b939d79dda1ab2bc17ad/postgres/src/main/java/com/google/cloud/sql/postgres/SocketFactory.java#L55-L62

Reply:
Oh I see. What about taking that suffix and passing it to an overload of `connect()`? Then the PostGres factory could call `connect(props, POSTGRES_SUFFIX)` and all the others could call `connect(props)` (where `connect(props)` would just call `connect(props, "")`).

That lets you keep the Postgres suffix in that module but do the environment-variable thing only once.

Reply:
I consider that vs my current approach and didn't really have a strong preference for one or the other. I'll refactor to that way if you think it's more reasonable.

Reply:
Done.